dist/truncate.js

```diff
@@ -72,19 +68,41 @@
     return false;
   }
 
-  /* Truncates the text content of a node using binary search.
-   * If no valid truncation point is found, attempt to truncate its nearest sibling.
-   *
-   * $textNode - The jQuery node to truncate.
-   * $rootNode - The jQuery root node to measure the truncated height.
-   * $clipNode - The jQuery node to insert right after the truncation point.
-   * options   - An object containing:
-   *             ellipsis  - The ellipsis string to append at the end of the truncation.
-   *             maxHeight - The maximum height for the root node.
-   *
-   * Returns true if truncation happened, false otherwise.
-   */
-  function truncateTextContent($element, $rootNode, $clipNode, options) { // jshint ignore:line
+  function truncateTextStart($element, $rootNode, $clipNode, options) {
+    var element = $element[0];
+    var original = $element.text();
+
+    var maxChunk = '';
+    var mid, chunk;
+    var low = 0;
+    var high = original.length;
+
+    while (low <= high) {
+      mid = low + ((high - low) >> 1); // Integer division
+
+      chunk = options.ellipsis + $.trim(original.substr(mid - 1, original.length));
+      setText(element, chunk);
+
+      if ($rootNode.height() > options.maxHeight) {
+        // too big, reduce the chunk
+        low = mid + 1;
+      }
+      else {
+        // chunk valid, try to get a bigger chunk
+        high = mid - 1;
+        maxChunk = maxChunk.length > chunk.length ? maxChunk : chunk;
+      }
+    }
+
+    if (maxChunk.length > 0) {
+      setText(element, maxChunk);
+      return true;
+    } else {
+      return truncateNearestSibling($element, $rootNode, $clipNode, options);
+    }
+  }
+
+  function truncateTextEnd($element, $rootNode, $clipNode, options) {
```

Comment:
It's a kind of private method, not like truncateTextContent or truncateNearestSibling which are the "entry points". I don't think that we should document them, I can add a comment to explain what is it, but I don't want to document them because a developper should not care about those methods. What do you think?